WalletWasabi.Gui/Global.cs

```diff
@@ -140,8 +139,11 @@ public async Task InitializeNoWalletAsync(TerminateService terminateService)
 
 				var userAgent = Constants.UserAgents.RandomElement();
 				var connectionParameters = new NodeConnectionParameters { UserAgent = userAgent };
+				UpdateChecker updateChecker = new(TimeSpan.FromMinutes(7), Synchronizer);
+
+				LegalChecker.InitializeAsync(updateChecker);
```

Comment:
It seems to me that `InitializeAsync` can be removed and you can use `LegalChecker` constructor. Then you don't need write:

`var content = await UpdateChecker!.WasabiClient ...` (that exclamation mark)

Reply:
You call async function without using await.

Reply:
I cannot init LegalChecker earlier as UpdateChecker created only here. On the other hand, LegalChecker passed forward earlier so I cannot move the ctor here either.